plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExBudgeting.hs

```diff
@@ -91,6 +103,73 @@ data CostModelBase f =
 deriving via CustomJSON '[FieldLabelModifier (StripPrefix "param", CamelToSnake)] (CostModelBase CostingFun) instance ToJSON (CostModelBase CostingFun)
 deriving via CustomJSON '[FieldLabelModifier (StripPrefix "param", CamelToSnake)] (CostModelBase CostingFun) instance FromJSON (CostModelBase CostingFun)
 
+{- Note [Cost model parameters]
+We want to expose to the ledger some notion of the "cost model parameters". Intuitively, these should be all the numbers that appear in the cost model.
+
+However, there are quite a few quirks to deal with.
+
+1. CostModel is stuctured
+
+That is, it's a complex data structure and the numbers in question are often nested inside it.
+To deal with this quickly, we take the ugly approach of operating on the JSON representation of the model.
+We flatten this down into a simple key-value mapping (see 'flattenObject' and 'unflattenObject'), and then
+look only at the numbers.
+
+2. We use floats, not integers
+
+We'd really prefer to expose integers as our parameters - they're just better behaved, and really we'd like to use integers
+internally too for determinism reasons. So we pretend that we have integers by scaling up all our numbers by 1000 and taking
+the integral floor, at some loss of precision.
+
+Once we use integers internally this will be simpler.
+
+3. CostModel includes the *type* of the model, which isn't a parameter
+
+We can just strip the out, but in particular this means that the parameters are not enough to *construct* a model.
+So we punt and say that you can *update* a model by giving the parameters. So you can take the default model and then
+overwrite the parameters, which seems okay.
+
+This is also implemented in a horrible JSON-y way.
+
+4. The implementation is not nice
+
+Ugly JSON stuff and failure possibilities where there probably shouldn't be any.
+-}
+
+-- See Note [Cost model parameters]
+type CostModelParams = Map.Map Text.Text Integer
+
+-- See Note [Cost model parameters]
+-- | Extract the model parameters from a model.
+extractModelParams :: CostModel -> Maybe CostModelParams
+extractModelParams cm = case toJSON cm of
+    Object o ->
+        let
+            flattened = flattenObject "-" o
+            toScaledInteger :: S.Scientific -> Integer
+            toScaledInteger n = floor (n*1000)
```

Comment:
Since 1000's used in two different places in might be worth lifting it to the top level.